src/jit/importer.cpp

```diff
@@ -11852,9 +11852,8 @@ MATH_MAYBE_CALL_NO_OVF:         ovfl = false;
                         //      3) Allocate and return the new object
                         // Reason: performance (today, we'll always use the slow helper for the R2R generics case)
 
-                        op1 = gtNewHelperCallNode(  info.compCompHnd->getNewHelper(&resolvedToken, info.compMethodHnd),
-                                                    TYP_REF, 0,
-                                                    gtNewArgList(op1));
+                        op1 = gtNewAllocObjNode(  info.compCompHnd->getNewHelper(&resolvedToken, info.compMethodHnd),
+                                                  resolvedToken.hClass, TYP_REF, op1 );
```

Comment:
This code always assigns the newly allocated object to a temp. However, it's not entirely clear that this is a fundamental requirement. So, either here or above the call to `impAssignTempGen`, I would add a comment that the `ObjectAllocator` relies on this.